MaterialLib/Component/CreateComponentProperties.h

```diff
@@ -0,0 +1,30 @@
+/**
+ * \copyright
+ * Copyright (c) 2012-2018, OpenGeoSys Community (http://www.opengeosys.org)
+ *            Distributed under a Modified BSD License.
+ *              See accompanying file LICENSE.txt or
+ *              http://www.opengeosys.org/project/license
+ *
+ * \file:   CreateComponentProperties.h
+ */
+
+#pragma once
+
+#include "BaseLib/ConfigTree.h"
```

Comment:
please use forward declarations of  ConfigTree,  ParameterBase and ComponentProperties, and move these includes to cpp file.

Reply:
Done. And also removed other unnecessary header files accordingly.